pyccl/halos/halo_model.py

```diff
@@ -988,6 +988,164 @@ def halomod_Tk3D_1h(cosmo, hmc,
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        lk_arr (array): an array holding values of the natural
+            logarithm of the wavenumber (in units of Mpc^-1) at
+            which the trispectrum should be calculated for
+            interpolation. If `None`, the internal values used
+            by `cosmo` will be used.
+        extrap_order_lok (int): extrapolation order to be used on
+            k-values below the minimum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        extrap_order_hik (int): extrapolation order to be used on
+            k-values above the maximum of the splines. See
+            :class:`~pyccl.tk3d.Tk3D`.
+        use_log (bool): if `True`, the trispectrum will be
+            interpolated in log-space (unless negative or
+            zero values are found).
+
+    Returns:
+        :class:`~pyccl.tk3d.Tk3D`: SSC effective trispectrum.
+    """
+    if lk_arr is None:
+        status = 0
+        nk = lib.get_pk_spline_nk(cosmo.cosmo)
+        lk_arr, status = lib.get_pk_spline_lk(cosmo.cosmo, nk, status)
+        check(status, cosmo=cosmo)
+    if a_arr is None:
+        status = 0
+        na = lib.get_pk_spline_na(cosmo.cosmo)
+        a_arr, status = lib.get_pk_spline_a(cosmo.cosmo, na, status)
+        check(status, cosmo=cosmo)
+
+    # Make sure biases are of the form number of a x number of k
+    ones = np.ones_like(a_arr)
+    bias1 *= ones
+    bias2 *= ones
+    bias3 *= ones
+    bias4 *= ones
+
+    k_use = np.exp(lk_arr)
+
+    # Check inputs
+    if not isinstance(prof, HaloProfileNFW):
+        raise TypeError("prof must be of type `HaloProfileNFW`")
+    prof_2pt = Profile2pt()
+
+    # Power spectrum
+    if isinstance(p_of_k_a, Pk2D):
+        pk2d = p_of_k_a
+    elif (p_of_k_a is None) or (str(p_of_k_a) == 'linear'):
+        pk2d = cosmo.get_linear_power('delta_matter:delta_matter')
+    elif str(p_of_k_a) == 'nonlinear':
+        pk2d = cosmo.get_nonlin_power('delta_matter:delta_matter')
+    else:
+        raise TypeError("p_of_k_a must be `None`, \'linear\', "
+                        "\'nonlinear\' or a `Pk2D` object")
+
+    na = len(a_arr)
+    nk = len(k_use)
+    dpk12 = np.zeros([na, nk])
+    dpk34 = np.zeros([na, nk])
+    for ia, aa in enumerate(a_arr):
+        # Compute profile normalizations
+        norm = hmc.profile_norm(cosmo, aa, prof) ** 2
+        i12 = hmc.I_1_2(cosmo, k_use, aa, prof, prof_2pt, prof) * norm
+
+        pk = pk2d.eval(k_use, aa, cosmo)
+        dpk = pk2d.eval_dlogpk_dlogk(k_use, aa, cosmo)
+        # ~ [(47/21 - 1/3 dlogPk/dlogk) * Pk+I12]
+        dpk12[ia] = ((2.2380952381-dpk/3)*pk + i12)
+        dpk34[ia] = dpk12[ia].copy()  # Avoid surprises
+
+        # Counter terms for clustering (i.e. - (bA + bB) * PAB
+        if is_number_counts1 or is_number_counts2 or is_number_counts3 or \
+           is_number_counts4:
+            b1 = b2 = b3 = b4 = 0
```

Comment:
you mentioned this can be dangerous. Is it only for arrays?

Reply:
Yes, it's only for arrays. In any case, to avoid modifying all of them, below I do `b1 = bias[ia]`, instead of `b1 += bias[ia]`